stl/inc/xmemory

```diff
@@ -547,16 +549,20 @@ struct _Normal_allocator_traits { // defines traits for allocators
     }
 #endif // _HAS_IF_CONSTEXPR
 
-    static void deallocate(_Alloc& _Al, pointer _Ptr, size_type _Count) {
+    static _CONSTEXPR20_DYNALLOC void deallocate(_Alloc& _Al, pointer _Ptr, size_type _Count) {
         _Al.deallocate(_Ptr, _Count);
     }
 
 #if _HAS_IF_CONSTEXPR
     template <class _Ty, class... _Types>
-    static void construct(_Alloc& _Al, _Ty* _Ptr, _Types&&... _Args) {
+    static _CONSTEXPR20_DYNALLOC void construct(_Alloc& _Al, _Ty* _Ptr, _Types&&... _Args) {
         if constexpr (_Uses_default_construct<_Alloc, _Ty*, _Types...>::value) {
             (void) _Al; // TRANSITION, DevCom-1004719
+#if _HAS_CXX20 && defined(__cpp_constexpr_dynamic_alloc)

```

Comment:
Please remove all occurrences of this pattern with `#ifdef __cpp_lib_constexpr_dynamic_alloc` (except for the one in yvals_core.h)

Reply:
Sorry for being that late, some vacation 😄. I will fix this pattern and the above reviews asap

Reply:
I would ask a maintainer, whether we want to replace the old functionality as done here or add an `if (_STD is_constant_evaluated())` branch

Reply:
The pattern has been replaced.
I will not resolve this one yet until a decision is made about using `if (_STD is_constant_evaluated())`